src/semgrep_agent/main.py

```diff
@@ -395,7 +396,7 @@ def protected_main(
             f"| to see your findings in the app, go to {publish_url}/manage/findings?repo={meta.repo_name}"
         )
 
-    exit_code = 1 if blocking_findings and not audit_mode else 0
+    exit_code = 0 if audit_mode else (2 if errors else 1 if blocking_findings else 0)
```

Comment:
I don't know how standard this is, but might be nice to document with a comment at the top of FindingSets where exit_code is defined.

Reply:
added some self-documenting code